students/jvirtue/lesson04/triagram.py

```diff
@@ -0,0 +1,90 @@
+#Lesson 4 Assignment 2
+#Triagrams
+#Jason Virtue 02/15/2019
+#UW Self Paced Python Course
+
+import random
+
+source = "sherlock_holmes.txt"
+
+def read_infile(source= source):
+    with open(source, "r") as infile:
+        book = infile.read()
+    return book
+
+def format_book(book):
+    book = book.replace("\n", " ")
+    book = book.replace("\\", " ")
+    words = book.split(" ")
+    while "" in words:
+        words.remove("")
+    for i in range(len(words)):
+        while " " in words[i]:
+            words[i] = words[i].replace(" ", "")
+        while "\\" in words[i]:
+            words[i] = words[i].replace("\\", "")
+        while "\\\'" in words[i]:
```

Comment:
no need for these `while` loops (lines 22, 24, 26). `replace` will replace all instances that you have without a loop.